Lib/test/test_asyncio/test_base_events.py

```diff
@@ -2214,17 +2214,48 @@ def check_set_nodelay(self, sock):
     @unittest.skipUnless(hasattr(socket, 'TCP_NODELAY'),
                          'need socket.TCP_NODELAY')
     def test_set_nodelay(self):
-        sock = socket.socket(family=socket.AF_INET, type=socket.SOCK_STREAM,
-                             proto=socket.IPPROTO_TCP)
-        with sock:
-            self.check_set_nodelay(sock)
+        with self.subTest('non-blocking'):
+            sock = socket.socket(family=socket.AF_INET, type=socket.SOCK_STREAM,
+                                 proto=socket.IPPROTO_TCP)
+            with sock:
+                self.check_set_nodelay(sock)
+
+        with self.subTest('blocking'):
+            sock = socket.socket(family=socket.AF_INET, type=socket.SOCK_STREAM,
+                                 proto=socket.IPPROTO_TCP)
+            with sock:
+                sock.setblocking(False)
+                self.check_set_nodelay(sock)
+
+
+    def check_set_quickack(self, sock):
+        # quickack already true by default on some OSes
+        opt = sock.getsockopt(socket.IPPROTO_TCP, socket.TCP_QUICKACK)
+        if opt:
+            base_events._set_quickack(sock, 0)
+
+        opt = sock.getsockopt(socket.IPPROTO_TCP, socket.TCP_QUICKACK)
+        self.assertFalse(opt)
 
-        sock = socket.socket(family=socket.AF_INET, type=socket.SOCK_STREAM,
-                             proto=socket.IPPROTO_TCP)
-        with sock:
-            sock.setblocking(False)
-            self.check_set_nodelay(sock)
+        base_events._set_quickack(sock, 1)
+
+        opt = sock.getsockopt(socket.IPPROTO_TCP, socket.TCP_QUICKACK)
+        self.assertTrue(opt)
 
+    @unittest.skipUnless(hasattr(socket, 'TCP_QUICKACK'),
+                         'need socket.TCP_QUICKACK')
+    def test_set_quickack(self):
+        with self.subTest('non-blocking'):
+            sock = socket.socket(family=socket.AF_INET, type=socket.SOCK_STREAM,
```

Comment:
Code looks duplicated. You can do:

```py
for blocking in (False, True):
    with self.subTest(blocking=blocking):
        ...
        sock.setblocking(blocking)
        ...
```

Reply:
Again, copying other tests.  I don't necessarily disagree, but if you want this kind of refactoring it really should be applied to the whole file.  I'm just trying to add a missing feature and using existing "guidelines" in what is already there.

Reply:
since I'm moving it anyway and it will no longer be a copy of an existing test, I'll implement this as well